src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java

```diff
@@ -150,8 +155,33 @@ protected AbstractEntityAIBasic(@NotNull final J job)
     @Override
     protected void onException(final RuntimeException e)
     {
-        Log.getLogger().info("Pausing Entity for 5 Seconds");
-        this.setDelay(EXCEPTION_TIMEOUT);
+        try
+        {
+            final int timeout = EXCEPTION_TIMEOUT * exceptionTimer;
+            this.setDelay(timeout);
+            // wait for longer now
+            exceptionTimer *= 2;
+            if (worker != null)
+            {
+                final String name = this.worker.getName();
+                final BlockPos workerPosition = worker.getPosition();
+                final AbstractJob colonyJob = worker.getColonyJob();
+                final String jobName = colonyJob == null ? "null" : colonyJob.getName();
+                Log.getLogger().error("Pausing Entity " + name + " (" + jobName + ") at " + workerPosition + " for " + timeout + " Seconds because of error:");
+            }
+            else
+            {
+                Log.getLogger().error("Pausing Entity that is null for " + timeout + " Seconds because of error:");
+            }
+
+            // fix for printing the actual exception
+            e.printStackTrace();
+        }catch (RuntimeException exp){
+            Log.getLogger().error("Welp reporting crashed:");
+            exp.printStackTrace();
+            Log.getLogger().error("Caused by ai exception:");
+            e.printStackTrace();
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Use a logger to log this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1148)